sdk/src/client/api/block_builder/mod.rs

```diff
@@ -4,34 +4,24 @@
 pub mod input_selection;
 pub mod transaction;
 
-use crypto::keys::bip44::Bip44;
-
 pub use self::transaction::verify_semantic;
 use crate::{
-    client::{
-        secret::{SecretManage, SignBlock},
-        ClientInner, Result,
-    },
+    client::{ClientInner, Result},
     types::block::{
-        core::{basic, BlockHeader, BlockWrapper},
+        core::{basic, BlockHeader, UnsignedBlock},
         payload::Payload,
         Block, IssuerId,
     },
 };
 
 impl ClientInner {
-    pub async fn build_basic_block<S: SecretManage>(
+    pub async fn build_basic_block(
         &self,
         issuer_id: IssuerId,
         issuing_time: Option<u64>,
         strong_parents: Option<basic::StrongParents>,
```

Comment:
Feels a bit weird that one can optionally define these, but not `commitment_id`, `weak_parents`, `shallow_like_parents`
We could change it to `Option<IssuanceBlockHeaderResponse>`, but at this point users can probably just build the whole thing themselves and don't need to call this function at all, so maybe just remove them instead?
Or do we see a common use case where it makes sense to manually specify these, but not the others?

Reply:
I don't disagree but I don't know what the best option is

Reply:
Simplified